src/Sylius/Bundle/ApiBundle/Canceller/PaymentRequestCanceller.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Canceller;
+
+use Sylius\Component\Payment\Model\PaymentRequestInterface;
+use Sylius\Component\Payment\Repository\PaymentRequestRepositoryInterface;
+
+/** @experimental */
+final class PaymentRequestCanceller implements PaymentRequestCancellerInterface
+{
+    /**
+     * @param PaymentRequestRepositoryInterface<PaymentRequestInterface> $paymentRequestRepository
+     */
+    public function __construct(private PaymentRequestRepositoryInterface $paymentRequestRepository)
+    {
+    }
+
+    public function cancelPaymentRequests(string $paymentId, string $paymentMethodCode): void
+    {
+        $paymentRequests = $this->paymentRequestRepository->findByStatesAndPaymentId([PaymentRequestInterface::STATE_NEW], $paymentId);
+
+        if ($paymentRequests === []) {
+            return;
+        }
+
+        /** @var PaymentRequestInterface $paymentRequest */
+        foreach ($paymentRequests as $paymentRequest) {
+            if ($paymentRequest->getMethod()->getCode() !== $paymentMethodCode) {
+                $paymentRequest->setState(PaymentRequestInterface::STATE_CANCELLED);
```

Comment:
I think `PaymentRequest`'s `$state` without state machine based state management needs ADR explanation

Reply:
I agree, we are actually planning to add state machine, just in another PR